lib/runtime/wasmer/instance.go

```diff
@@ -216,6 +216,50 @@ func (in *Instance) UpdateRuntimeCode(code []byte) error {
 	return nil
 }
 
+// CheckRuntimeVersion calculates runtime Version for runtime blob passed in
+func (in *Instance) CheckRuntimeVersion(code []byte) (runtime.Version, error) {
+	originalVM := in.vm
+
+	imports, err := in.imports()
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: determine memory descriptor size that the runtime wants from the wasm.
+	// should be doable w/ wasmer 1.0.0.
+	memory, err := wasm.NewMemory(23, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = imports.AppendMemory("memory", memory)
+	if err != nil {
+		return nil, err
+	}
+
+	// Instantiates the WebAssembly module.
+	instance, err := wasm.NewInstanceWithImports(code, imports)
+	if err != nil {
+		return nil, err
+	}
+
+	// Assume imported memory is used if runtime does not export any
+	if !instance.HasMemory() {
+		instance.Memory = memory
+	}
+
+	instance.SetContextData(in.ctx)
+
+	in.vm = instance
+	checkVersion, err := in.Version()
```

Comment:
not sure why this is needed, you can just call `instance.Version()` directly and remove the `originalVM := in.vm` and `in.vm = originalVM` lines

Reply:
I can't call `instance.Version()`  because instance is a wasmer instance which doesn't have Version() function.